### PR TITLE
Update Go vers, x/sys vendor

### PR DESCRIPTION
**Pull request by estesp**
Update linter version and align linter settings with main containerd project.

### CODE REVIEW

**Review by austinvazquez — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	github.com/stretchr/testify v1.8.0
```
Comment (thaJeztah) [inline go.mod:6]: Not urgent, but I saw there was a v1.8.1 as well

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
Ah; linter complains;

```
prefix=src/github.com/containerd/fifo] in [/Users/runner/work/fifo/fifo/src/github.com/containerd/fifo] ...
  Error: directive `//nolint: unconvert` is unused for linter "unconvert" (nolintlint)
  
  Error: issues found
```

**Review by thaJeztah — APPROVED** [review]
still LGTM

**Review by mikebrow — APPROVED** [review]
LGTM

**Review by dmcgowan — APPROVED** [review]

**Pull request merged**